ingest.py

```diff
@@ -1,112 +1,133 @@
-        "https://python.langchain.com/docs/modules", 
-        "https://python.langchain.com/docs/guides",
-        "https://python.langchain.com/docs/ecosystem",
-        "https://python.langchain.com/docs/additional_resources",
-        "https://python.langchain.com/docs/community",
+        # "https://python.langchain.com/docs/integrations",
+        # "https://python.langchain.com/docs/modules",
+        # "https://python.langchain.com/docs/guides",
+        # "https://python.langchain.com/docs/ecosystem",
+        # "https://python.langchain.com/docs/additional_resources",
+        # "https://python.langchain.com/docs/community",
     ]
 
     documents = []
     for url in urls:
-        loader = RecursiveUrlLoader(url=url, max_depth=2 if url == urls[0] else 10, extractor=lambda x: Soup(x, "lxml").text, prevent_outside=True)
+        loader = RecursiveUrlLoader(
+            url=url,
+            max_depth=10,
+            extractor=lambda x: Soup(x, "lxml").text,
+            prevent_outside=True,
+        )
         temp_docs = loader.load()
-        temp_docs = [doc for i, doc in enumerate(temp_docs) if doc not in temp_docs[:i]]        
+        temp_docs = [
+            doc
+            for i, doc in enumerate(temp_docs)
+            if doc not in temp_docs[:i] and doc not in documents
+        ]
         documents += temp_docs
         print("Loaded", len(temp_docs), "documents from", url)
-    
     print("Loaded", len(documents), "documents from all URLs")
-    
     html2text = Html2TextTransformer()
-    docs_transformed = html2text.transform_documents(documents)
-    
-    repo_docs = ingest_repo()
-    docs_transformed += repo_docs
-    
-    print("Transformed", len(docs_transformed), "documents in total")
-    
-    # OPTION TO PICKLE
-    # import pickle
-    # with open('docs_transformed.pkl', 'wb') as f:
-    #     pickle.dump(docs_transformed, f)
-    # with open('docs_transformed.pkl', 'rb') as f:
-    #     docs_transformed = pickle.load(f)
-    
-    client = weaviate.Client(url=WEAVIATE_URL, auth_client_secret=weaviate.AuthApiKey(api_key=WEAVIATE_API_KEY))
-    child_splitter = RecursiveCharacterTextSplitter(chunk_size=1000)
-    parent_splitter = RecursiveCharacterTextSplitter(chunk_size=2000)
-    client.schema.delete_class("LangChain_parents_idx") # delete the class if it already exists
-    
+    return list(html2text.transform_documents(documents))
+
+
+def _value_serializer(value) -> str:
+    if isinstance(value, Document):
+        value = {
+            "page_content": value.page_content,
+            "metadata": value.metadata,
+        }
+    return json.dumps(value)
+
+
+def _value_deserializer(serialized_value: str) -> Document:
+    value = json.loads(serialized_value)
+    if "page_content" in value and "metadata" in value:
+        return Document(page_content=value["page_content"], metadata=value["metadata"])
+    else:
+        return value
+
+
+def _index_docs(
```

Comment:
meat of it: do parent/child splitting explicitly (not relying or ParentDocumentRetriever) and then running indexing